pkg/controller/plan/kubevirt.go

```diff
@@ -239,6 +242,11 @@ func (r *KubeVirt) EnsureVM(vm *plan.VMStatus) (err error) {
 		err = liberr.Wrap(err)
 		return
 	}
+	//If the VM name is not valid according to DNS1123 labeling
+	//convention it will be automatically changed.
+	if len(k8svalidation.IsDNS1123Label(vm.Name)) > 0 {
+		newVM.Name = changeVmNameDNS1123(vm.Name, vm.ID)
+	}
```

Comment:
i think this part should be within the `virtualMachine()`

Reply:
yeah, makes sense to either move it there or change the vm before calling `virtualMachine(...)`

Reply:
agreed, moved to `virtualMachine()`